libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -408,16 +414,46 @@ export class CartService {
       customer = await this.customerManager.retrieve(cart.stripeCustomerId);
     }
 
-    const invoicePreview = await this.invoiceManager.preview({
+    const upcomingInvoicePreview = await this.invoiceManager.previewUpcoming({
       priceId: price.id,
       customer,
       taxAddress: cart.taxAddress || undefined,
       couponCode: cart.couponCode || undefined,
     });
 
+    // Cart latest invoice data
+    let latestInvoicePreview: InvoicePreview | undefined;
+    let last4: string | undefined;
+    if (customer && cart.stripeSubscriptionId) {
+      // fetch latest payment info from subscription
+      const subscription = await this.subscriptionManager.retrieve(
+        cart.stripeSubscriptionId
+      );
+      assert(subscription.latest_invoice, 'Subscription not found');
+      latestInvoicePreview = await this.invoiceManager.preview(
+        subscription.latest_invoice
+      );
+
+      // fetch payment method info
+      if (subscription.collection_method === 'send_invoice') {
+        // PayPal payment method collection
+        // TODO: render paypal payment info in the UI
+      } else {
+        // Stripe payment method collection
+        if (subscription.default_payment_method) {
```

Comment:
Could this be switched to `customer.invoice_settings.default_payment_method`? Seems like that's what's used most commonly

Reply:
Good idea, updating to use the suggested